src/ert/_c_wrappers/enkf/ecl_config.py

```diff
@@ -1,207 +1,96 @@
-
-            # REFCASE_KEY
-            refcase_default = config_dict.get(ConfigKeys.REFCASE)
-            if refcase_default is not None:
-                refcase_default = os.path.realpath(refcase_default)
-
-            # SCHEDULE_PREDICTION_FILE_KEY
-            schedule_prediction_file = config_dict.get(
-                ConfigKeys.SCHEDULE_PREDICTION_FILE
+        if ConfigKeys.GRID in config_dict:
+            ecl_config_args["grid_file"] = os.path.realpath(
+                config_dict[ConfigKeys.GRID]
             )
-
-            c_ptr = self._alloc_full(
-                have_eclbase,
-                data_file,
-                grid,
-                refcase_default,
-                schedule_prediction_file,
+        if ConfigKeys.REFCASE in config_dict:
+            ecl_config_args["refcase_file"] = os.path.realpath(
+                config_dict[ConfigKeys.REFCASE]
             )
-            if grid is not None:
-                grid.convertToCReference(None)
-
-        if c_ptr:
-            super().__init__(c_ptr)
-        else:
-            raise RuntimeError("Internal error: Failed constructing EclConfig!")
-
-    def free(self):
-        self._free()
-
-    def getDataFile(self):
-        return self._get_data_file()
-
-    def setDataFile(self, datafile):
-        self._set_data_file(datafile)
-
-    def validateDataFile(self, datafile) -> UIReturn:
-        """@rtype: UIReturn"""
-        return self._validate_data_file(datafile)
-
-    # -----------------------------------------------------------------
-
-    def get_gridfile(self) -> str:
-        return self._get_gridfile()
-
-    def set_gridfile(self, gridfile):
-        self._set_gridfile(gridfile)
-
-    def validateGridFile(self, gridfile):
-        return self._validate_gridfile(gridfile)
-
-    def getGrid(self) -> EclGrid:
-        return self._get_grid()
-
-    def getRefcaseName(self):
-        return self._get_refcase_name()
-
-    def loadRefcase(self, refcase):
-        self._load_refcase(refcase)
-
-    def getRefcase(self) -> EclSum:
-        refcase = self._get_refcase()
-        if refcase is not None:
-            refcase.setParent(self)
-
-        return refcase
-
-    def validateRefcase(self, refcase):
-        return self._validate_refcase(refcase)
-
-    def hasRefcase(self):
-        """@rtype: bool"""
-        return self._has_refcase()
-
-    # -----------------------------------------------------------------
-
-    def getDepthUnit(self):
-        return self._get_depth_unit()
-
-    def getPressureUnit(self):
-        return self._get_pressure_unit()
-
-    # -----------------------------------------------------------------
-
-    def active(self):
-        """
-        Has ECLIPSE been configured?"
-        """
-        return self._active()
-
-    def getLastHistoryRestart(self):
-        return self._get_last_history_restart()
+        return cls(**ecl_config_args)
+
+    def _load_grid(grid_file: str) -> Optional[EclGrid]:
```

Comment:
`def _load_grid(self, grid_file: str) -> Optional[EclGrid]`

Reply:
sorry, that should have been a class method (to make it clear that it can be moved outta there)

Reply:
never mind, making it a stand alone function, see above